test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -115,8 +118,18 @@ def hook(context):
 
 @OnScenarioEnd
 def hook(context):
-    # capture screenshot if there is error in the scenario execution, and if the test is being run in CI
-    if test.resultCount("errors") > 0 and os.getenv('CI'):
+    # This workaround is needed because resultCount() return number of results of the given resultCategory(fails/errors) of the current test cases/ tests suite.
```

Comment:
```suggestion
    # Currently, this workaround is needed because we cannot find out a way to determine the pass/fail status of currently running test scenario.
    # And, resultCount("errors")  and resultCount("fails") return the total number of error/failed test scenarios of a test suite.
```